### ISSUE TITLE
Different Results When Running Multiple SQL Statements Together vs. Individually in ClickHouse

### ISSUE THREAD

**Issue by awusan125** [ClickHouse/ClickHouse#70824]
**Describe what's wrong**

The result is different when running these four statements together compared to running them one by one.

When run together, it returns an empty string.
```sql
   ┌─c1─┐
1. │    │
   └────┘

1 row in set. Elapsed: 0.001 sec. 
```
 When run one by one, it returns 1. 
```sql

   ┌─c1─┐
1. │ 1  │
   └────┘

1 row in set. Elapsed: 0.002 sec. 
```

**How to reproduce**
```sql
DROP TABLE IF EXISTS t0;

CREATE TABLE t0 (c0 String, c1 String) ENGINE = MergeTree() ORDER BY tuple();

INSERT INTO t0 (c0) VALUES ('0'), ('');

ALTER TABLE t0 UPDATE c1 = '1' WHERE c0 IN ('0', '');

SELECT c1 FROM t0 WHERE c0 = '';
```

**Expected behavior**

They should return the same result.

**Comment by den-crane** [comment]
Clickhouse version? https://fiddle.clickhouse.com/73a30db8-68a1-4242-a76c-50fa3c6ed641

BTW `ALTER TABLE t0 UPDATE` is eventual, you need to use `settings mutations_sync=2` to make the repro stable.

**Comment by awusan125** [comment]
```sql
734b673d1b7c :) select version();

SELECT version()



Query id: 24fc87d1-b014-4079-beee-9535212a938b

   ┌─version()─┐
1. │ 24.9.2.42 │
   └───────────┘

1 row in set. Elapsed: 0.001 sec. 

734b673d1b7c :) 
```

**Comment by den-crane** [comment]
@awusan125 I am unable to reproduce the issue https://fiddle.clickhouse.com/f8ff793f-1385-4bce-b45f-620f67aceb47

**Comment by akuluasan** [comment]
Yes. When `settings mutations_sync=2` is added, the two ways run the same. The example I tested did not add `settings mutations_sync=2`.https://fiddle.clickhouse.com/da0c0b27-dccb-4598-b1d1-32636d78af05

**Comment by den-crane** [comment]
That is expected
https://clickhouse.com/docs/en/sql-reference/statements/alter/update The synchronicity of the query processing is defined by the [mutations_sync](https://clickhouse.com/docs/en/operations/settings/settings#mutations_sync) setting. By default, it is asynchronous.
